Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/BucketDetectionSystem.cs

```diff
@@ -14,19 +14,26 @@ protected override void OnUpdate()
 
         var helperEntity = GetSingletonEntity<FreeBucketInfo>();
         var bucketBuffer = EntityManager.GetBuffer<FreeBucketInfo>(helperEntity);
+        var frame = GetCurrentFrame();
         
         bucketBuffer.Clear();
         
         Entities.WithAll<BucketTag>()
             .WithName("BucketInfoCollect")
             .ForEach((in Entity entity, in MyBucketState state, in Position position) =>
             {
-                switch (state.Value)
+                // do not scoop buckets on same frame -- too many race conditions there.
+                if (frame > state.FrameChanged + 2)
```

Comment:
Do you mean that the system shouldn't deal with bucket dropped during the current frame?